src/nunavut/lang/c/support/serialization.j2

```diff
@@ -0,0 +1,556 @@
+{%- macro choose_min(a, b) -%}
+    ((({{a}}) < ({{b}})) ? ({{a}}) : ({{b}}))
+{%- endmacro -%}
+/*
+ *                                                                                                            +-+ +-+
+ *  UAVCAN common serialization support routines.                                                             | | | |
+ *                                                                                                            \  -  /
+ *  This software is distributed under the terms of the MIT License.                                            ---
+ *                                                                                                               o
+ * +------------------------------------------------------------------------------------------------------------------+
+ */
+#ifndef NUNAVUT_SUPPORT_SERIALIZATION_H_INCLUDED
+#define NUNAVUT_SUPPORT_SERIALIZATION_H_INCLUDED
+
+#ifdef __cplusplus
+#   if (__cplusplus < 201100L)
+#       error "Unsupported language: ISO C11, C++11, or a newer version of either is required."
+#   endif
+extern "C"
+{
+#else
+#   if !defined(__STDC_VERSION__) || (__STDC_VERSION__ < 201112L)
+#       error "Unsupported language: ISO C11 or a newer version is required."
+#   endif
+#endif
+
+#include <string.h>
+{%- if not option_omit_float_serialization_support %}
+#include <float.h>
+{% endif -%}
+#include <stdbool.h>
+#include <stdint.h>
+
```

Comment:
Ack, sorry. I lead you astray. we'll need assert.h in C11 for the static_assert macro:

```
#ifndef __cplusplus
#include <assert.h>
#endif
```